firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,544 @@
+        """
+
+        active_update = self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            return ums.activeUpdate;
+
+        """)
+
+        return bool(active_update)
+
+    def get_patch_at(self, patch_index):
+        """Use nsIUpdate.getPatchAt to return a patch from an update
+
+        :returns: nsIUpdatePatch object
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            let patch_index = arguments[0];
+            return active_update.getPatchAt(patch_index);
+
+        """, script_args=[patch_index])
+
+    @property
+    def patch_count(self):
+        """Get the patchCount from the active update
+
+        :returns: The patch count
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.patchCount;
+
+        """)
+
+    @property
+    def selected_patch_final_url(self):
+        """Get the finalURL of the selected patch for the active update
+
+        :returns: The patch size
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.finalURL;
+
+        """)
+
+    @property
+    def selected_patch_size(self):
+        """Get the size of the selected patch for the active update
+
+        :returns: The patch size
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.size;
+
+        """)
+
+    @property
+    def selected_patch_type(self):
+        """Get the type of the selected patch for the active update
+
+        :returns: The patch type
+        """
+
+        return self.marionette.execute_script("""
+            let ums = Cc['@mozilla.org/updates/update-manager;1']
+                .getService(Ci.nsIUpdateManager);
+            let active_update = ums.activeUpdate;
+            return active_update.selectedPatch.type;
+
+        """)
+
+
+class MARChannels(object):
+    """Class to handle the allowed MAR channels as listed in update-settings.ini
+    """
+
+    INI_SECTION = 'Settings'
+    INI_OPTION = 'ACCEPTED_MAR_CHANNEL_IDS'
+
+    REGEX_UPDATE_CHANNEL_PREF = re.compile(r'("app\.update\.channel", ")([^"].*)(?=")')
+
+    def __init__(self, software_update):
```

Comment:
This class should be based on BaseLib instead of object, so it would require the marionette getter as parameter to the __init__ method.